components/apps.js

```diff
@@ -649,33 +647,151 @@ SteamUser.prototype.getOwnedDepots = function(excludeSharedLicenses) {
+	return this.getOwnedDepots(filter).indexOf(parseInt(depotid, 10)) != -1;
 };
 
 /**
- * Returns an array of package IDs this account owns. Only works if enablePicsCache option is enabled and appOwnershipCached event
- * has been emitted.
- * @param {boolean} [excludeSharedLicenses=false] - Pass true to exclude licenses that we have through family sharing
+ * Returns an array of licenses this account owns.
  * @returns {int[]}
  */
-SteamUser.prototype.getOwnedPackages = function(excludeSharedLicenses) {
-	if (this.steamID.type != SteamID.Type.ANON_USER && !this.licenses) {
+SteamUser.prototype.getOwnedLicenses = function() {
+	if (!this.licenses) {
 		throw new Error("We don't have our license list yet.");
 	}
 
+	return this.licenses;
+};
+
+/**
+ * Returns an array of package IDs this account owns (different from owned licenses). The filter only
+ * works, if enablePicsCache option is enabled and appOwnershipCached event has been emitted.
+ * @param {object|function} filter - Options for what counts for ownership, or a custom filter function
+ * @returns {int[]}
+ */
+SteamUser.prototype.getOwnedPackages = function(filter) {
+	// We're anonymous
 	if (this.steamID.type == SteamID.Type.ANON_USER) {
 		return [17906];
 	}
 
 	// We're an individual user
-	let packages = this.licenses;
-	if (excludeSharedLicenses) {
-		packages = packages.filter(license => license.owner_id == this.steamID.accountid);
+	let packages = this.getOwnedLicenses();
+
+	// If passed filter argument is empty, and global ownership filter is set, use it
+	if (!filter && this.options.ownershipFilter) {
+		filter = this.options.ownershipFilter;
+	}
+
+	// Support for the old syntax and behavior (<= v4.19.12)
+	if (!filter) {
+		// Slight deviation of old behavior: don't return expired licenses
+		return this._returnPackages(packages);
+	}
+	// If filter options is a boolean, we asssume it's excludeSharedLicenses
+	if (typeof filter === 'boolean') {
+		let excludeSharedLicenses = filter;
+		filter = {excludeShared: excludeSharedLicenses};
+	}
+
+	// New behavior from this point on
+	// Handle filter options
+	if (typeof filter === 'object') {
+		const defaults = {
+			excludeFree: false, // By default, include free licenses (FreeOnDemand, NoCost, etc.)
+			excludeShared: false, // By default, include shared licenses
+			excludeExpiring: false // By default, include licenses that are going to expire in the future (free weekends)
+		};
+		filter = Object.assign(defaults, this.options.ownershipFilter, filter);
+
+		// If there is nothing to filter, no pics cache needed
+		if (!filter.excludeFree && !filter.exludeShared && !filter.excludeExpiring) {
+			return this._returnPackages(packages);
+		}
+
+		// We don't need pics cache for only excludeShared
+		if (!filter.excludeFree && filter.exludeShared && !filter.excludeExpiring) {
+			packages = packages.filter(license => license.owner_id == this.steamID.accountid);
+			return this._returnPackages(packages);
+		}
+	}
```

Comment:
```js
// no PICS cache needed
if (!filter.excludeFree && !filter.excludeExpiring) {
	if (filter.exludeShared) {
		packages = packages.filter(license => license.owner_id == this.steamID.accountid);
	}
	return this._returnPackages(packages);
}
```

Reply:
Also, because this part line https://github.com/DoctorMcKay/node-steam-user/blob/18c1a8bab4fcf7992193a6a7c2e9e23cc308dad2/components/apps.js#L770 will never be reached.

Reply:
I don't think this is true, if you have filter options {F=1,S=1,E=0} or {F=1,S=0,E=0}, you can reach it. Right?
Check my latest commit

Reply:
Oh, I mean `expirytime` check is unreachable.

> if you have filter options {F=1,S=1,E=0} or {F=1,S=0,E=0}, you can reach it. Right?

No. It will be filtered by `excludeFree` as `owned && ...` is always `false`.
If I am wrong, do you know any non-free packages with `expirytime` field?

Reply:
Not sure. But do you know if `pkg.extended.expirytime <= Math.floor(Date.now() / 1000);` is covered by `!(license.flags & SteamUser.ELicenseFlags.Expired)`?

Reply:
It is, but I don't know if flag is set immediately after `expirytime` or some time later. And you will need a fresh `licenses` list, which Steam may sent to you a few hours after the end of `FreeWeekend`.

Reply:
Actually I found some mismatches. Also found out why. `RegionRestrictionExpired` flag also falls under `Expired`.

Reply:
How should I handle this?

Reply:
This flag applies only 3 months after purchase. This means that *some* country restrictions (on the package) were removed, and you [probably] can run the game in another country. It is not about license expiration.

Reply:
Can you come to discord to discuss this? https://discord.gg/wYwmPbjP